### PR TITLE
Fix Unsort and raise error

### LINKED ISSUE #79: [BUG] Fix unsorting of data in Ordered weight L1 Norm and also ensure we error out in case we use increasing weights
**Describe the bug**

1) 

https://github.com/CEA-COSMIC/ModOpt/blob/621598738a5048dc72f3baa3e11f06f2071a65bc/modopt/opt/proximity.py#L521-L523

The above codes is supposed to unsort the data however this is wrong in current state.

I will move this to as proposed by @zaccharieramzi  in https://github.com/CEA-COSMIC/ModOpt/pull/52#discussion_r337008989

2) 

https://github.com/CEA-COSMIC/ModOpt/blob/621598738a5048dc72f3baa3e11f06f2071a65bc/modopt/opt/proximity.py#L479

Here the code reorders weights, which causes issues:
a) It prevents users from providing constant weight number and not a float.
b) Results in bad and undesired output if weights arent non-increasing as required by theory of this operator.

I will get a PR for both of these.

### PR DESCRIPTION
**Pull request by chaithyagr**
This fixes #79

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ branches:
 # install package and dependencies
 install:
   - pip install coverage nose pytest pytest-cov
+  - pip install scikit-learn
```
Comment (sfarrens) [inline .travis.yml]: Why was this dependency added?

File: modopt/opt/proximity.py [hunk 2]
```diff
@@ -475,8 +475,9 @@ def __init__(self, weights):
                               ' found see documentation for details: '
                               'https://cea-cosmic.github.io/ModOpt/'
                               '#optional-packages')
-
-        self.weights = np.sort(weights.flatten())[::-1]
+        if np.max(np.diff(weights)) > 0:
+            raise ValueError('Weights must be non increasing')
```
Comment (sfarrens) [inline modopt/opt/proximity.py]: As a user, I am not sure would understand what `'Weights must be non increasing'` means. Perhaps you can rephrase this. *e.g.* "the weight values must be provided in descending order" or something like that.

File: .travis.yml (hunk 1 above)
Comment (chaithyagr) [reply .travis.yml]: Yes, I wanted to talk to you about this.
This is not a dependency for the codes, but if this is added, Travis actually tests all the OWL norm codes. This bug was missed especially as scikit learn was not installed and none of OWL norm tests ran.
scikit-learn in itself will continue to remain an optional dependency

File: modopt/opt/proximity.py (hunk 2 above)
Comment (chaithyagr) [reply modopt/opt/proximity.py]: Done

**Pull request merged**